src/lib/holocene/modal.svelte

```diff
@@ -91,29 +120,35 @@
           <h3>Title</h3>
         </slot>
       </div>
-      <div id="modal-content" class="content">
-        <slot name="content">
-          <span>Content</span>
-        </slot>
-      </div>
-      <div class="flex items-center justify-end space-x-2 p-6">
-        <Button
-          thin
-          variant="secondary"
-          disabled={loading}
-          on:click={cancelModal}>{cancelText}</Button
-        >
-        {#if !hideConfirm}
+      <svelte:element
+        this={hasInput ? 'form' : 'div'}
```

Comment:
Do the actions need to be inside the `<form/>`? Could we just pass the form into the `<slot/>` for the modal? 
e.g.
```js
<Modal on:confirm={handleConfirm}>
    <form id="example-form" on:submit={handleConfirm}>
      <input/>
   </form>
</Modal>
```

And then have some way of making the confirm button `type="submit"` and have a reference to the form's `id`?
E.g.
```js
<Button type={formId ? "submit" : "button"} form={formId} />`

Reply:
Oh nice I didn't know this was a thing. I think we definitely should have done it this way if I hadn't done a large refactor 😂 Now `<Modal />` uses a `<dialog />` element which always contains a `<form />`, and the form always contains the buttons. I implemented it largely according to [this example](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog#simple_example).